jax/_src/numpy/lax_numpy.py

```diff
@@ -607,12 +607,14 @@ def power(x1, x2):
 
   # TODO(phawkins): add integer pow support to XLA.
   bits = 6  # Anything more would overflow for any x1 > 1
-  acc = ones(shape(x1), dtype=dtype)
+  zero = _constant_like(x2, 0)
+  one = _constant_like(x2, 1)
+  # Initialize acc carefully such that pow(0, x2) is zero for x2 != 0
```

Comment:
I'm not 100% sure it's necessary that 0**0 is 1, but it's not the worst idea, so let's keep it.

Reply:
It's necessary if we want to match numpy's output in this case

Reply:
That's my question: is this an intentional behavior of NumPy or an accidental behavior of its implementation? We certainly do not want to match NumPy for all inputs, e.g., if integer overflow occurs.  That said, matching numpy for this specific input seems fine.

Reply:
I think there are good arguments to define `0^0 = 1`, particularly in the context of integers. Searching around, Knuth argues this way:

> Some textbooks leave the quantity 0^0 undefined, because the functions x^0 and 0^x have different limiting values when x decreases to 0. But this is a mistake. We must define x0 = 1, for all x, if the binomial theorem is to be valid when x = 0, y = 0, and/or x = −y. The binomial theorem is too important to be arbitrarily restricted! By contrast, the function 0^x is quite unimportant"

Reply:
Another good argument is that IEEE 754-2008 section 9.2.1 specifies pow(0, 0) == 1.